### PR TITLE
[ONNX] Unify ONNXProgram and remove the old one

### PR DESCRIPTION
**Pull request by justinchuby**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #136277
* #136335
* #136279
* __->__ #136281

## Note

`test_fx_to_onnx_with_onnxruntime.py` is removed for now (it has a lot of xfails anyways). A better version will be added back.

Fixes https://github.com/pytorch/pytorch/issues/136274

cc @albanD

### CODE REVIEW

**Review by xadupre — APPROVED** [review]

**Review by albanD — APPROVED** [review]
Skipping PR sanity check for code removal sounds good!
Thanks for the cleanup!

File: torch/onnx/_internal/_exporter_legacy.py [hunk 1]
```diff
@@ -1023,13 +608,8 @@ def export(self) -> ONNXProgram:
...
                 )
 
-            return torch.onnx.ONNXProgram(
-                onnx_model,
-                self.options.fx_tracer.input_adapter,
-                self.options.fx_tracer.output_adapter,
-                self.options.diagnostic_context,
-                fake_context=self.options.fake_context,
-                model_torch=self.model,
+            return _onnx_program.ONNXProgram(
```
Comment (titaiwangms) [inline torch/onnx/_internal/_exporter_legacy.py:611]: @justinchuby How do we call io_adapter in this scenario?

**Pull request closed without merging**